### PR TITLE
(maint) Remove `paged_results.*` wrapper around generated sql

### PR DESCRIPTION
**Pull request by ajroetker**
This commit makes some syntatical changes to the query engine paging
functions as well as removing the `paged_results` wrapper and the sql we
generate for query-engine queries as it is no longer necessary.

### CODE REVIEW

**Comment by wkalt** [comment]
+1

File: src/puppetlabs/puppetdb/query/paging.clj [hunk 1]
```diff
@@ -216,17 +216,8 @@
...
 (defn dealias-order-by
   [{:keys [projections] :as query-rec} paging-options]
-  (let [alias-map (reduce-kv (fn [acc k v]
-                               (assoc acc
-                                 (keyword k)
-                                 (-> v :field h/extract-sql)))
-                             {} projections)]
-    (update-in paging-options [:order_by]
-               #(map (partial rename-first alias-map) %))))
+  (let [alias-map (->> (kitchensink/mapvals (comp h/extract-sql :field) projections)
```
Comment (mullr) [inline src/puppetlabs/puppetdb/query/paging.clj]: 1. Is this a refactor? if so, it needs to be in its own commit. 
2. Threading with two things is kind of pointless and a bit confusing, would much rather see

    (->> projections 
      (kitchensink/mapvals (comp h/extract-sql :field))
      (kitchensink/mapkeys keyword))

File: src/puppetlabs/puppetdb/query/paging.clj [hunk 2]
```diff
@@ -216,17 +216,8 @@
...
+                       (kitchensink/mapkeys keyword))]
+    (update paging-options :order_by (partial map #(update % 0 (partial get alias-map))))))
```
Comment (mullr) [inline src/puppetlabs/puppetdb/query/paging.clj]: I have no idea what this line does... is it the same as it used to be? if so, I like the old one better.

File: src/puppetlabs/puppetdb/query/paging.clj (hunk 1 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/paging.clj]: Why 1? Also I see your point with 2 and will change but threading with two things isn't necessarily pointless

Comment (mullr) [inline src/puppetlabs/puppetdb/query/paging.clj]: 1 because otherwise figuring out if something is intended to be a refactor or a functional change is like solving a little puzzle every time

Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/paging.clj]: But this whole PR is framed as a maint/refactor

Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/paging.clj]: I'd agree that if there we're a functional change it would need to be in its own commit, maybe even a note about it in the commit message, but a separate commit I'm not sure about

Comment (mullr) [inline src/puppetlabs/puppetdb/query/paging.clj]: for the generated sql. This is incidental to that, yes?

Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/paging.clj]: Same codepath, the goal is to eventually have this pipeline refactored so that we don't need to keywordize strings and then turn them back to strings.

Comment (mullr) [inline src/puppetlabs/puppetdb/query/paging.clj]: probably shouldn't have picked this time and place to start this discussion. But: especially for someone with imperfect knowledge of the codebase, isolating changes as much as possible to small, interdependent sets that leave the system in a working state is very valuable. Each of them can have its own commit message, in this case 'simplify dealias-order-by' would have been sufficient. That would let any reviewer plus a future history delver know the intent behind the change, which is like 75% of why we do source control in the first place. 

I know this isn't generally what we do, and that I'm asking for a change. I also know that github doesn't support this well. But I feel like we need to figure something out here.

**Comment by ajroetker** [comment]
@mullr updated the commit message at least

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
transient ^
